kernel/mod_subst.ml

```diff
@@ -427,6 +427,13 @@ let subst_mps subst c =
   if is_empty_subst subst then c
   else map_kn (subst_mind subst) (subst_pcon_term subst) c
 
+let subst_mps_list substs c =
+  if List.is_empty substs || List.for_all is_empty_subst substs then c
+  else
+    let subst_const pcon = List.fold_right subst_mps substs (mkConstU pcon) in
```

Comment:
won't this produce different results with inlining?

Reply:
I guess the second commit changes it back but then maybe they should be squashed so that we don't have an incorrect commit around

Reply:
Why would it generate different results even in presence of inlining? We're recursively expanding the term generated so this is the same?